### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by scurker**
Allows sync pull requests to be created on pull request merges. [Working example here](https://github.com/dequelabs/github-actions-playground/pull/39).

Features:
* Set the head/base branches (defaults to `master`/`develop`) 
* Can set the title, body, label, reviewers and assignees of the pr
* Can default to using the default PR template of the repo (could not find any other action that supports this)

It's dockerized unfortunately, because [Github apparently needs node_modules to be committed to the repository](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-a-javascript-action#commit-and-push-your-action-to-github) and doesn't run any kind of install steps. 🤷‍♂ 

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+version: 2.1
```
Comment (stephenmathieson) [inline .circleci/config.yml]: I'm rather surprised you didn't use GitHub actions for this 🙃

File: README.md [hunk 2]
```diff
@@ -1 +1,47 @@
...
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          pr-title: "chore: merge master into develop"
+          pr-reviewers: scurker
+          pr-labels: chore
+```
+
+## Configuration
+
+- **github-token** (required) - needed to interact with github api
```
Comment (stephenmathieson) [inline README.md]: This is provided by default, right? Or do we need to generate a special token just for this tool?

File: src/main.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      ...context.repo,
+      issue_number: pullNumber,
+      assignees: assignees.split(',')
+    })
+  }
+}
+
+// Prevent action from auto-running in test environment
+if (process.env.NODE_ENV !== 'test') {
+  run().catch(function(err) {
```
Comment (stephenmathieson) [inline src/main.ts]: Arrow function?

File: .circleci/config.yml (hunk 1 above)
Comment (scurker) [reply .circleci/config.yml]: I wanted to, but I thought as a team we were still using circle for most of our jobs?

File: README.md (hunk 2 above)
Comment (scurker) [reply README.md]: Yup, github provides a [short lived temporary access token](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token) only for this repo.

File: .circleci/config.yml (hunk 1 above)
Comment (stephenmathieson) [reply .circleci/config.yml]: Yes, we do. For consistency, this is probably better 🤷‍♂

File: README.md (hunk 2 above)
Comment (stephenmathieson) [reply README.md]: Can we mention this in the config documentation then? The way this reads suggests to me that I need to _provide_ a `github-token` value.

Comment (scurker) [reply README.md]: You still have to pass it into the action:

> To use the `GITHUB_TOKEN` secret, you must reference it in your workflow file

Comment (stephenmathieson) [reply README.md]: Makes sense 👍

**Pull request merged**